extensions/src/org/pentaho/platform/web/http/api/resources/services/UserRoleListService.java

```diff
@@ -71,7 +71,17 @@ public UserListWrapper getUsers() {
   }
 
   public RoleListWrapper getRoles() {
-    return new RoleListWrapper( getUserRoleListService().getAllRoles() );
+    return getRoles( true );
+  }
+
+  public RoleListWrapper getRoles( boolean includeExtraRoles ) {
+    List<String> roles = getUserRoleListService().getAllRoles();
+    if ( !includeExtraRoles ) {
+      for ( String role : getExtraRoles() ) {
+        roles.remove( role );
```

Comment:
I suppose it should be left as-is. We just need to document that they cannot have a regular role anywhere that is the same as an "extra" role.